packages/paper-qa-pymupdf/tests/test_paperqa_pymupdf.py

```diff
@@ -258,3 +274,16 @@ def custom_to_markdown(self, clean=False, fill_empty=True) -> str:
             "\n|**2.0** <br>|3/6 (50%)|2/6 (33%)|1/6 (17%)|"
             "\n\n"  # NOTE: this is before strip, so there can be trailing whitespace
         )
+
+
+def test_equation_parsing() -> None:
+    parsed_text = parse_pdf_to_pages(STUB_DATA_DIR / "duplicate_media.pdf")
+    assert isinstance(parsed_text.content, dict)
+    assert isinstance(parsed_text.content["1"], tuple)
+    p1_text, p1_media = parsed_text.content["1"]
+    # SEE: https://regex101.com/r/pyOHLq/1
+    assert re.search(
+        r"[_*]*E[_*]* ?= ?[_*]*mc[_*]*(?:<sup>)?[ ^]?[2²] ?(?:<\/sup>)?", p1_text
+    ), "Expected inline equation in page 1 text"
```

Comment:
This doesn't cover LaTeX macros, see the last case here: https://regex101.com/r/1p2wpa/1

Maybe try this one? https://stackoverflow.com/questions/14182879/regex-to-match-latex-equations 
https://regex101.com/r/wP2aV6/25

Reply:
So for context, this assertion is a superset of what our the readers (PyPDF, PyMuPDF, Docling, Nemotron Parse) will return per this inline PDF content: 

<img width="217" height="40" alt="screenshot of duplicate media PDF" src="https://github.com/user-attachments/assets/8bfd8943-d9aa-49ae-bf62-d9103c952a40" />

That `\sqrt{m^2 c^4 + p^2 c^2}` case or the LaTeX macro for block formula case won't come up. And if they did come up, it would be a PDF misread of the inline LaTeX.

In general this assertion is a 'sanity check' that we're reading the formula. If new cases arrive, we can adjust then as needed.

Let me know what you're thinking, but I don't think we need to cover LaTeX macros

Reply:
Ah ok, this is just for rendered PDFs, got it. Sounds good!